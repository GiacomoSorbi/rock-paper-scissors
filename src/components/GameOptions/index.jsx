import React from 'react'
import './GameOptions.css'
import Button from '../Button'

const GameOptions = ({ options, onClick }) => (
  <div className='options-container'>
    {options.map(option => (
      <Button key={option} onClick={onClick} name={option} />
    ))}
  </div>
)

export default GameOptions
