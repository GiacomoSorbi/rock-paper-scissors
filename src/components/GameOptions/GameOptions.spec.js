import React from 'react'
import { render, fireEvent, waitForElement } from '@testing-library/react'
// import 'jest-dom/extend-expect'
import GameOptions from './index'

describe('GameOptions should', () => {
  it('render', () => {
    const options = ['pippi', 'cat', 'pappa']
    const { asFragment } = render(<GameOptions options={options} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('render buttons passed in the props', () => {
    const options = ['pippi', 'cat', 'pappa']
    const fn = jest.fn()
    const { getByText } = render(<GameOptions options={options} />)
    for (let i = 0; i < options.length; i++) {
      const clickedElement = options[i]

      expect(getByText(clickedElement))
    }
  })

  it('render buttons that trigger the passed `onClick` prop', () => {
    const options = ['pippi', 'cat', 'pappa']
    const fn = jest.fn()
    const { getByText } = render(<GameOptions options={options} onClick={fn} />)

    const clickedElement = options[0]
    fireEvent.click(getByText(clickedElement))

    expect(fn).toHaveBeenCalledTimes(1)
  })
})
