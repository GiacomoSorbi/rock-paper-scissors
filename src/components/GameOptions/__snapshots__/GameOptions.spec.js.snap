// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameOptions should render 1`] = `
<DocumentFragment>
  <div>
    <button
      id="pippi"
    >
      pippi
    </button>
    <button
      id="cat"
    >
      cat
    </button>
    <button
      id="pappa"
    >
      pappa
    </button>
  </div>
</DocumentFragment>
`;
